scripts/updateDatabase.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+## This script will create or replace a database using a mysql dump file. It takes the DB name (-d or --database) and dump filepath (-f or --file) as arguments.
+## A config file populated with mySQL username, password and host is also required. Its filepath can be optionally specified (-c or --config), otherwise a default 'config.properties' file is assumed.
+## author: jcook
+
+dbFilepath=
+dbName=
+configFilepath=config.properties
+## Parse command-line arguments
+while (( "$#" )); do
+	case "$1" in
+		-f|--file)
+			dbFilepath=$2;
+			shift 2;
+			;;
+		-d|--database)
+			dbName=$2;
+			shift 2;
+			;;
+		-c|--config)
+			configFilepath=$2;
+			shift 2;
+			;;
+		-*|--*=)
+			echo "Error: Unsupported flag $1"
+			exit 1
+	esac
+done
+
+## If missing arguments, explain usage
+if [ -z "$dbFilepath" ] || [ -z "$dbName" ]
+then
+	echo "Create or replace mySQL databases";
+	echo "Usage: bash updateDatabase.sh -f databaseFile -d databaseName [-c configFile] ";
+	exit 1
+fi
+
+## Parse config file for username and password properties
+echo "Reading $configFilepath";
+username=
+password=
+host=localhost
+port=3306
+while read line; do
+	if [[ $line == username* ]]
+ 	then
+ 		username=${line#*=}
+ 	elif [[ $line == password* ]]
+ 	then
+ 		password=${line#*=}
+	elif [[ $line == host* ]]
+	then
+		host=${line#*=}
+	elif [[ $line == port* ]]
+	then
+		port=${line#*=}
+ 	fi
+done < $configFilepath
+
+## Improperly formatted or missing config file
+if [ -z "$username" ] || [ -z "$password" ]
+then
+	echo "Could not find username and/or password in $configFilepath";
+	exit 1
+fi
+
+echo "Updating $dbName with $dbFilepath";
+echo "Backing up $dbName";
+
+## Take archive of DB, drop it and create a new, empty one
+dbArchiveFile="$dbName.backup.dump"
+mysqldump -u$username -p$password -h$host -P$port $dbName > $dbArchiveFile
+echo "Compressing $dbArchiveFile";
+eval "gzip -f $dbArchiveFile";
+echo "Finished backing up $dbName";
+
+echo "Updating $dbName with $dbFilepath";
+mysql -u$username -p$password -h$host -P$port -e 'drop database if exists $dbName; create database $dbName'
+
+## If dump is gzipped, must use 'zcat'
```

Comment:
Would it run faster if you unzip it and then `cat` the unzipped file, vs. using zcat? Is zcat always installed or should that be checked first?

Reply:
I can't say that I've timed the two, though it doesn't take very long either way. And I know zcat is installed in base linux though I just found that there might be alternative behaviour for zcat on macs: https://serverfault.com/questions/570024/zcat-gzcat-works-in-linux-not-on-osx-general-linux-osx-compatibility

Reply:
hmm ok probably not a problem to leave it this way for now since the _real_ release server runs linux (not OS X). BUT... if it's possible to find a way of calling zcat that is compatible for both platforms that would be nice to have, but not necessary right now.

Reply:
Yea that makes sense. Alternatively I could just unzip the file if it has the *.gz pattern, I would probably just want to zip it up again at the end -- which would add a bit of time if it was a very large dump file

Reply:
Hmm good point about re-zipping at the end. OK, I guess just leave it as is for now.